odl/space/base_tensors.py

```diff
@@ -258,6 +259,21 @@ def astype(self, dtype):
         else:
             return self._astype(dtype)
 
+    def asweighted(self, weighting):
```

Comment:
How about this? It's going to be needed at some point, but not strictly here.

Another idea with this functionality would be to stuff it inside `astype` in `kwargs`, and to extend that function with more stuff that we might want to change. For instance interpolation, `impl`, or even trivial stuff like axis labels.

Reply:
Bumping this